benches/rsexp_benchmark.rs

```diff
@@ -0,0 +1,67 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use rand::seq::SliceRandom;
+use rand::thread_rng;
+
+fn make_n_random_characters(n: i64, alphabet: &Vec<char>) -> String {
+    let mut rng = thread_rng();
+    (0..n).map(|_| alphabet.choose(&mut rng).unwrap()).collect()
+}
+
+fn make_benchmark_string(
+    num_repetitions: &[i64],
+    str_len: i64,
+    quoted: bool,
+    alphabet: &Vec<char>,
+) -> String {
+    if let Some(len) = num_repetitions.get(0) {
+        format!(
+            "({})",
+            std::iter::repeat(make_benchmark_string(
+                &num_repetitions[1..],
+                str_len,
+                quoted,
+                alphabet
+            ))
+            .take(*len as usize)
+            .collect::<String>()
+        )
+    } else {
+        let chars = make_n_random_characters(str_len, alphabet);
+        if quoted {
+            format!("\"{}\"", chars)
+        } else {
+            chars
+        }
+    }
+}
+
+fn parse_sexp(contents: &[u8]) {
+    let _sexp = rsexp::from_slice(&contents).unwrap();
+}
+
+pub fn criterion_benchmark(c: &mut Criterion) {
```

Comment:
Should the rng be seeded always in the same way for this function? This would make the benchmark slightly easier to understand though I doubt that it would have an impact as it only matters for the alphabet chars.

Reply:
Good point, added a seed below and passed it down to where it's used.